tests/check_parachain_lease_expiration.ts

```diff
@@ -0,0 +1,78 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import '@polkadot/api-augment';
+import '@polkadot/api-augment/kusama';
+import '@polkadot/api-augment/polkadot';
+import { assert } from 'chai';
+
+export async function createPromiseApi(nodeAddress: string) {
+  const wsProvider = new WsProvider(nodeAddress);
+
+  const api = new ApiPromise({
+    provider: wsProvider,
+  });
+  await api.isReady;
+  console.log(`${nodeAddress} has been started`);
+  return api;
+}
+
+describe('Check Parachain_Lease_Expiration', () => {
+    it('Check Manta Parachain Lease', async () => {
+        const polkadotNode = 'wss://1rpc.io/dot';
+        const polkadotApi = await createPromiseApi(polkadotNode);
+        const mantaParaId = 2104;
+        const oneDay = 3600 * 24;
+
+        const relaychainBlockTime = polkadotApi.consts.babe.expectedBlockTime.toNumber() / 1000;
+        const bestNumber = (await polkadotApi.derive.chain.bestNumber()).toNumber();
+        const startNumber = bestNumber - polkadotApi.consts.slots.leaseOffset.toNumber();
+        const leasePeriod = polkadotApi.consts.slots.leasePeriod.toNumber();
+        const currentPeriod = Math.floor(startNumber / leasePeriod);
+
+        const remainingLeaseLength = await polkadotApi.query.slots.leases(mantaParaId);
+        const endPeriod = currentPeriod + remainingLeaseLength.length - 1;
+
+        await polkadotApi.disconnect();
+        assert(remainingLeaseLength.length >= 1, 'Manta lease is going to expire, please do auction.');
+        console.log(`Remaining lease period for manta: ${currentPeriod}-${endPeriod}`);
+
+        const firstBlock = startNumber % leasePeriod;
+        if (firstBlock === 0) {
+            const remainingDays = leasePeriod * relaychainBlockTime * remainingLeaseLength.length / oneDay;
+            console.log(`Manta parachain lease has ${remainingDays} days remaining.`);
+        } else {
+            const currentLeaseRemainedBlocks = leasePeriod - firstBlock;
+            const remainingDays = leasePeriod * relaychainBlockTime * (remainingLeaseLength.length - 1) / oneDay + (currentLeaseRemainedBlocks * relaychainBlockTime / oneDay);
+            console.log(`Manta parachain lease has ${remainingDays} days remaining.`);
+        }
+    }).timeout(2000000);
```

Comment:
whats the timeout number for, maybe declare this as a named var?

Reply:
I think the problem is the test framework `mocha` doesn't support `async` well, so I have to add a timeout for each test case. And `2000` sec is unreasonable, now it's 5 minutes.